aspnetcore/blazor/debug.md

```diff
@@ -1,123 +1,180 @@
+     ![Additional setup requried](https://devblogs.microsoft.com/aspnet/wp-content/uploads/sites/16/2020/03/vscode-additional-setup.png)
 
-Blazor provides a debugging proxy that implements the [Chrome DevTools Protocol](https://chromedevtools.github.io/devtools-protocol/) and augments the protocol with .NET-specific information. When debugging keyboard shortcut is pressed, Blazor points the Chrome DevTools at the proxy. The proxy connects to the browser window you're seeking to debug (hence the need to enable remote debugging).
+   * A notification offers to add the required assets to the app for building and debugging. Select **Yes**:
+
+     ![Add required assets](https://devblogs.microsoft.com/aspnet/wp-content/uploads/sites/16/2020/03/vscode-required-assets.png)
+
+1. Starting the app in the debugger is a two-step process:
+
+   1\. **First**, start the app using the **.NET Core Launch (Blazor Standalone)** launch configuration.
+
+   2\. **After the app has started**, start the browser using the **.NET Core Debug Blazor Web Assembly in Chrome** launch configuration (requires Chrome). To use Edge instead of Chrome, change the `type` of the launch configuration in *.vscode/launch.json* from `pwa-chrome` to `pwa-edge`.
+
+1. Set a breakpoint in the `IncrementCount` method in the `Counter` component and then select the button to hit the breakpoint:
 
-## Attach to process in Visual Studio
+   ![Debug Counter in VS Code](https://devblogs.microsoft.com/aspnet/wp-content/uploads/sites/16/2020/03/vscode-debug-counter.png)
 
-Attaching to the app's process in Visual Studio is a *temporary* debugging scenario for Blazor WebAssembly while **F5** debugging is in development.
+### Debug in the browser
 
-To attach the running app's process to Visual Studio:
+1. Run a Debug build of the app in the Development environment.
 
-1. In Visual Studio, select **Debug** > **Attach to Process**.
-1. For the **Connection type**, select **Chrome devtools protocol websocket (no authentication)**.
-1. For the **Connection target**, paste in the HTTP address (not the HTTPS address) of the app.
-1. Select **Refresh** to refresh the entries under **Available processes**.
-1. Select the browser process to debug and select **Attach**.
-1. In the **Select Code Type** dialog, select the code type for the specific browser you're attaching to (Edge or Chrome) and then select **OK**.
+1. Press <kbd>Shift</kbd>+<kbd>Alt</kbd>+<kbd>D</kbd>.
+
+1. Follow the instructions to restart the browser with [remote debugging enabled](#enable-remote-debugging), if needed.
+
+1. Use the browser's developer tools to debug the app.
+
+## Enable remote debugging
+
+If remote debugging is disabled, an **Unable to find debuggable browser tab** error page is generated by Chrome. The error page contains instructions for running Chrome with the debugging port open so that the Blazor debugging proxy can connect to the app. *Close all Chrome instances* and restart Chrome as instructed.
```

Comment:
```suggestion
To debug a Blazor WebAssembly app the browser must be run with remote debugging enabled. If remote debugging is disabled, an **Unable to find debuggable browser tab** error page is generated. The error page contains instructions for running the browser with the debugging port open so that the Blazor debugging proxy can connect to the app. *Close all browser instances* and restart the browser as instructed.
```